model/PermissionsService.php

```diff
@@ -132,11 +132,24 @@ private function wetRun(array $actions): void
         if(!empty($actions['remove'])){
             $this->dataBaseAccess->removeMultiplePermissions($actions['remove']);
         }
-        if(!empty($actions['add'])){
-            $this->dataBaseAccess->addMultiplePermissions($actions['add']);
+        if(!empty($actions['add'])) {
+            $this->dataBaseAccess->addMultiplePermissions(
+                $this->removeDuplicatedActions($actions['add'])
+            );
         }
     }
 
+    private function removeDuplicatedActions(array $addActions): array
```

Comment:
Should we not change `getActions` method instead to avoid duplications? https://github.com/oat-sa/extension-tao-dac-simple/pull/197/files#diff-040d5dec37d951ce87553fa9c4ad8eaa53185d0b9057d3e6d43c4acf98dbbbceR96

Reply:
Actually we need to discover the root cause, why we started having double entries in this array

Reply:
That was my initial thought. We may do that, but then I think we'll need to use an additional array or something similar that would allow to directly query the actions per resource ID (or iterate all values already added to `$actions['remove']` and `$actions['add']`, which may be a bit slow (and the code for that a bit ugly :sweat_smile: ))